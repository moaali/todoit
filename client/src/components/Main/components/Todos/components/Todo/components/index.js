export { default as Checkbox } from './Checkbox';
export { default as Controls } from './Controls';
export { default as EditBody } from './EditBody';
export { default as ViewBody } from './ViewBody';
