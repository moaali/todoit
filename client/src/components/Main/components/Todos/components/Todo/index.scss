@import 'settings/index';
@import 'tools/index';
@import 'helpers/index';

.todo {
  // Box Model
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-between;
  padding: 30px;
  min-height: 360px;
  max-height: 360px;
  height: 50vmax;

  // Cosmetics
  background: white;
  border-radius: 4px;
  box-shadow: 0 0 3px 3px rgba($color-primary, 0.04);

  .offcanvas & {
    // Box Model
    padding: 0;
    height: auto;

    // Cosmetics
    border-radius: 0;
    box-shadow: none;
  }
}

.todo--is-completed {
  // Cosmetics
  border: 2px solid $color-success;
}
