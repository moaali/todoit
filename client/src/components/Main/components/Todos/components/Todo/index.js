import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { inject } from 'mobx-react'

import { Checkbox, Controls, EditBody, ViewBody } from './components'
import './index.scss';

@inject('store')
class Todo extends PureComponent {
  static propTypes = {
    store: PropTypes.shape({
      offcanvasStore: PropTypes.shape({}).isRequired,
      todosStore: PropTypes.shape({}).isRequired,
    }).isRequired,
    todo: PropTypes.shape({}).isRequired,
  }

  constructor(props) {
    super(props);

    this.todosStore = props.store.todosStore;
    this.offcanvasStore = props.store.offcanvasStore;
    this.state = {
      isEditMode: false,
      todo: props.todo,
    }
  }

  handleCheck = event => {
    this.todosStore.updateTodo({
      ...this.state.todo,
      completed: event.target.checked,
    })

    this.setState(prevState => ({
      ...prevState,
      todo: {
        ...prevState.todo,
        completed: !prevState.todo.completed,
      },
    }));
  }

  handleTitleChange = event => {
    const { value } = event.target;

    this.setState(prevState => ({
      ...prevState,
      todo: {
        ...prevState.todo,
        title: value,
      },
    }));
  }

  handleDecriptionChange = event => {
    const { value } = event.target;

    this.setState(prevState => ({
      ...prevState,
      todo: {
        ...prevState.todo,
        description: value,
      },
    }));
  }

  handleEditMode = () => {
    this.setState(prevState => ({
      isEditMode: !prevState.isEditMode,
    }));
  }

  handleView = () => {
    this.todosStore.viewTodo(this.state.todo.id);
    this.offcanvasStore.toggleOffcanvasVisibility();
  }

  renderBody = todo => {
    if (this.state.isEditMode) {
      return (
        <EditBody
          handleTitleChange={this.handleTitleChange}
          handleDecriptionChange={this.handleDecriptionChange}
          todo={todo}
        />
      );
    }

    return <ViewBody todo={todo} />
  }

  render() {
    const { todo, isEditMode } = this.state;

    return (
      <div className={`todo ${todo.completed ? 'todo--is-completed' : ''}`}>
        <div className="peers jc-sb w-100p">
          <div className="peer">
            <Checkbox todo={todo} handleCheck={this.handleCheck} />
          </div>
          <div className="peer">
            <Controls
              isEditMode={isEditMode}
              handleEditMode={this.handleEditMode}
              handleView={this.handleView}
              todo={todo}
            />
          </div>
        </div>
        <div className='w-100p'>
          <div className='mT-40'>
            { this.renderBody(todo) }
          </div>
        </div>
      </div>
    );
  }
}

export default Todo;
