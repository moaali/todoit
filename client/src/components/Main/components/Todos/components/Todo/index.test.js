import React from 'react'
import { Provider } from 'mobx-react';

import stores from 'stores';
import Todo from './'

const todo = {
  id: '1',
  completed: true,
  title: 'Todo Title',
  description: 'Todo description.',
  date: 'Jan 01, 2020',
}

const TodoComponent = () => (
  <Provider store={{...stores}}>
    <Todo todo={todo} />
  </Provider>
);

describe('Todo', () => {
  it('should match snapshot', () => {
    const component = mount(<TodoComponent />);
    expect(toJson(component)).toMatchSnapshot();
  })
})
