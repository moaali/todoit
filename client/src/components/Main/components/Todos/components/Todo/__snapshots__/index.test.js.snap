// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should match snapshot 1`] = `
<TodoComponent>
  <Provider
    store={
      Object {
        "offcanvasStore": OffcanvasStore {},
        "searchStore": SearchStore {},
        "sidebarStore": SidebarStore {},
        "todosStore": TodosStore {
          "activeCategory": "all",
          "applyFilters": [Function],
          "currentOffset": 0,
          "displayedTodos": Array [],
          "filterByCategory": [Function],
          "filterByCompleted": [Function],
          "filterByQuery": [Function],
          "filters": Object {
            "category": "all",
            "completed": null,
            "query": "",
          },
          "findById": [Function],
          "loadChunkSize": 10,
          "loadTodos": [Function],
          "meta": Object {
            "all": 0,
            "completed": 0,
            "pending": 0,
          },
          "requestStatus": "error",
          "shownTodo": null,
          "todos": Array [],
        },
      }
    }
  >
    <inject-Todo-with-store
      todo={
        Object {
          "completed": true,
          "date": "Jan 01, 2020",
          "description": "Todo description.",
          "id": "1",
          "title": "Todo Title",
        }
      }
    >
      <Todo
        store={
          Object {
            "offcanvasStore": OffcanvasStore {},
            "searchStore": SearchStore {},
            "sidebarStore": SidebarStore {},
            "todosStore": TodosStore {
              "activeCategory": "all",
              "applyFilters": [Function],
              "currentOffset": 0,
              "displayedTodos": Array [],
              "filterByCategory": [Function],
              "filterByCompleted": [Function],
              "filterByQuery": [Function],
              "filters": Object {
                "category": "all",
                "completed": null,
                "query": "",
              },
              "findById": [Function],
              "loadChunkSize": 10,
              "loadTodos": [Function],
              "meta": Object {
                "all": 0,
                "completed": 0,
                "pending": 0,
              },
              "requestStatus": "error",
              "shownTodo": null,
              "todos": Array [],
            },
          }
        }
        todo={
          Object {
            "completed": true,
            "date": "Jan 01, 2020",
            "description": "Todo description.",
            "id": "1",
            "title": "Todo Title",
          }
        }
      >
        <div
          className="todo todo--is-completed"
        >
          <div
            className="peers jc-sb w-100p"
          >
            <div
              className="peer"
            >
              <inject-Checkbox-with-store
                handleCheck={[Function]}
                todo={
                  Object {
                    "completed": true,
                    "date": "Jan 01, 2020",
                    "description": "Todo description.",
                    "id": "1",
                    "title": "Todo Title",
                  }
                }
              >
                <Checkbox
                  handleCheck={[Function]}
                  store={
                    Object {
                      "offcanvasStore": OffcanvasStore {},
                      "searchStore": SearchStore {},
                      "sidebarStore": SidebarStore {},
                      "todosStore": TodosStore {
                        "activeCategory": "all",
                        "applyFilters": [Function],
                        "currentOffset": 0,
                        "displayedTodos": Array [],
                        "filterByCategory": [Function],
                        "filterByCompleted": [Function],
                        "filterByQuery": [Function],
                        "filters": Object {
                          "category": "all",
                          "completed": null,
                          "query": "",
                        },
                        "findById": [Function],
                        "loadChunkSize": 10,
                        "loadTodos": [Function],
                        "meta": Object {
                          "all": 0,
                          "completed": 0,
                          "pending": 0,
                        },
                        "requestStatus": "error",
                        "shownTodo": null,
                        "todos": Array [],
                      },
                    }
                  }
                  todo={
                    Object {
                      "completed": true,
                      "date": "Jan 01, 2020",
                      "description": "Todo description.",
                      "id": "1",
                      "title": "Todo Title",
                    }
                  }
                >
                  <label
                    className="checkbox checkbox--big"
                    htmlFor="checkbox__input-1"
                  >
                    <input
                      className="checkbox__input"
                      defaultChecked={true}
                      id="checkbox__input-1"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="checkbox__checkmark"
                    />
                  </label>
                </Checkbox>
              </inject-Checkbox-with-store>
            </div>
            <div
              className="peer"
            >
              <Controls
                handleEditMode={[Function]}
                handleView={[Function]}
                isEditMode={false}
                todo={
                  Object {
                    "completed": true,
                    "date": "Jan 01, 2020",
                    "description": "Todo description.",
                    "id": "1",
                    "title": "Todo Title",
                  }
                }
              >
                <ul
                  className="peers todo-controls"
                >
                  <inject-Delete-with-store
                    todo={
                      Object {
                        "completed": true,
                        "date": "Jan 01, 2020",
                        "description": "Todo description.",
                        "id": "1",
                        "title": "Todo Title",
                      }
                    }
                  >
                    <Delete
                      store={
                        Object {
                          "offcanvasStore": OffcanvasStore {},
                          "searchStore": SearchStore {},
                          "sidebarStore": SidebarStore {},
                          "todosStore": TodosStore {
                            "activeCategory": "all",
                            "applyFilters": [Function],
                            "currentOffset": 0,
                            "displayedTodos": Array [],
                            "filterByCategory": [Function],
                            "filterByCompleted": [Function],
                            "filterByQuery": [Function],
                            "filters": Object {
                              "category": "all",
                              "completed": null,
                              "query": "",
                            },
                            "findById": [Function],
                            "loadChunkSize": 10,
                            "loadTodos": [Function],
                            "meta": Object {
                              "all": 0,
                              "completed": 0,
                              "pending": 0,
                            },
                            "requestStatus": "error",
                            "shownTodo": null,
                            "todos": Array [],
                          },
                        }
                      }
                      todo={
                        Object {
                          "completed": true,
                          "date": "Jan 01, 2020",
                          "description": "Todo description.",
                          "id": "1",
                          "title": "Todo Title",
                        }
                      }
                    >
                      <li
                        className="peer todo-controls__item"
                      >
                        <a
                          className="fa fa-trash todo-controls__icon"
                          data-for="controls__control-tip-delete-1"
                          data-tip={true}
                          href="#"
                          onClick={[Function]}
                        >
                           
                        </a>
                        <ReactTooltip
                          effect="solid"
                          id="controls__control-tip-delete-1"
                          insecure={true}
                          resizeHide={true}
                          wrapper="div"
                        >
                          <div
                            className="__react_component_tooltip place-top type-dark "
                            data-id="tooltip"
                            id="controls__control-tip-delete-1"
                          >
                            <span>
                              Delete Todo
                            </span>
                          </div>
                        </ReactTooltip>
                      </li>
                    </Delete>
                  </inject-Delete-with-store>
                  <inject-View-with-store
                    handleView={[Function]}
                    isEditMode={false}
                    todo={
                      Object {
                        "completed": true,
                        "date": "Jan 01, 2020",
                        "description": "Todo description.",
                        "id": "1",
                        "title": "Todo Title",
                      }
                    }
                  >
                    <View
                      handleView={[Function]}
                      isEditMode={false}
                      store={
                        Object {
                          "offcanvasStore": OffcanvasStore {},
                          "searchStore": SearchStore {},
                          "sidebarStore": SidebarStore {},
                          "todosStore": TodosStore {
                            "activeCategory": "all",
                            "applyFilters": [Function],
                            "currentOffset": 0,
                            "displayedTodos": Array [],
                            "filterByCategory": [Function],
                            "filterByCompleted": [Function],
                            "filterByQuery": [Function],
                            "filters": Object {
                              "category": "all",
                              "completed": null,
                              "query": "",
                            },
                            "findById": [Function],
                            "loadChunkSize": 10,
                            "loadTodos": [Function],
                            "meta": Object {
                              "all": 0,
                              "completed": 0,
                              "pending": 0,
                            },
                            "requestStatus": "error",
                            "shownTodo": null,
                            "todos": Array [],
                          },
                        }
                      }
                      todo={
                        Object {
                          "completed": true,
                          "date": "Jan 01, 2020",
                          "description": "Todo description.",
                          "id": "1",
                          "title": "Todo Title",
                        }
                      }
                    >
                      <li
                        className="peer mL-10 todo-controls__item"
                      >
                        <a
                          className="fa fa-eye todo-controls__icon"
                          data-for="controls__control-tip-view-1"
                          data-tip={true}
                          href="#"
                          onClick={[Function]}
                        >
                           
                        </a>
                        <ReactTooltip
                          effect="solid"
                          id="controls__control-tip-view-1"
                          insecure={true}
                          resizeHide={true}
                          wrapper="div"
                        >
                          <div
                            className="__react_component_tooltip place-top type-dark "
                            data-id="tooltip"
                            id="controls__control-tip-view-1"
                          >
                            <span>
                              View Todo
                            </span>
                          </div>
                        </ReactTooltip>
                      </li>
                    </View>
                  </inject-View-with-store>
                  <inject-Edit-with-store
                    handleEditMode={[Function]}
                    todo={
                      Object {
                        "completed": true,
                        "date": "Jan 01, 2020",
                        "description": "Todo description.",
                        "id": "1",
                        "title": "Todo Title",
                      }
                    }
                  >
                    <Edit
                      handleEditMode={[Function]}
                      store={
                        Object {
                          "offcanvasStore": OffcanvasStore {},
                          "searchStore": SearchStore {},
                          "sidebarStore": SidebarStore {},
                          "todosStore": TodosStore {
                            "activeCategory": "all",
                            "applyFilters": [Function],
                            "currentOffset": 0,
                            "displayedTodos": Array [],
                            "filterByCategory": [Function],
                            "filterByCompleted": [Function],
                            "filterByQuery": [Function],
                            "filters": Object {
                              "category": "all",
                              "completed": null,
                              "query": "",
                            },
                            "findById": [Function],
                            "loadChunkSize": 10,
                            "loadTodos": [Function],
                            "meta": Object {
                              "all": 0,
                              "completed": 0,
                              "pending": 0,
                            },
                            "requestStatus": "error",
                            "shownTodo": null,
                            "todos": Array [],
                          },
                        }
                      }
                      todo={
                        Object {
                          "completed": true,
                          "date": "Jan 01, 2020",
                          "description": "Todo description.",
                          "id": "1",
                          "title": "Todo Title",
                        }
                      }
                    >
                      <li
                        className="peer mL-10 todo-controls__item"
                      >
                        <a
                          className="fa fa-pencil todo-controls__icon"
                          data-for="controls__control-tip-edit-1"
                          data-tip={true}
                          href="#"
                          onClick={[Function]}
                        >
                           
                        </a>
                        <ReactTooltip
                          effect="solid"
                          id="controls__control-tip-edit-1"
                          insecure={true}
                          resizeHide={true}
                          wrapper="div"
                        >
                          <div
                            className="__react_component_tooltip place-top type-dark "
                            data-id="tooltip"
                            id="controls__control-tip-edit-1"
                          >
                            <span>
                              Edit Todo
                            </span>
                          </div>
                        </ReactTooltip>
                      </li>
                    </Edit>
                  </inject-Edit-with-store>
                </ul>
              </Controls>
            </div>
          </div>
          <div
            className="w-100p"
          >
            <div
              className="mT-40"
            >
              <ViewBody
                todo={
                  Object {
                    "completed": true,
                    "date": "Jan 01, 2020",
                    "description": "Todo description.",
                    "id": "1",
                    "title": "Todo Title",
                  }
                }
              >
                <i
                  className="op-50p"
                >
                  Jan 01, 2020
                </i>
                <h2
                  className="mB-20 mT-10 tt-c"
                >
                  Todo Title
                </h2>
                <p>
                  Todo description.
                </p>
              </ViewBody>
            </div>
          </div>
        </div>
      </Todo>
    </inject-Todo-with-store>
  </Provider>
</TodoComponent>
`;
